plugins/google-sheets/src/sheets.ts

```diff
@@ -515,19 +534,56 @@ export async function getPluginContext(): Promise<PluginContext> {
     // turns out to be false too often - switch to batchGetByDataFilter +
     // GridRange and drop titles altogether.
 
+    const sheetHeaderRow = sheet.values[0]
+    const currentSheetHeaderRowHash = generateHashId(sheetHeaderRow.join(","))
```

Comment:
Hmm, re-ordering columns shouldn't trigger a re-sync. To avoid this, I think this needs to be sorted before hashing?

Also, maybe use a spicier delimiter, that's unlikely to occur in source data. Could just do a constant random string, e.g., `ErSW0wuIm7KM90LF`, since we aren't length-limited.

`generateHashId` is also a poor checksum hash, but it's out of scope of this PR 😄 Just noting it for awareness.

Reply:
> re-ordering columns shouldn't trigger a re-sync

Or, maybe it's fine! It's not a huge price to pay, and column re-ordering doesn't happen every day.